### PR TITLE
DM-8420: Wrap coadd_utils with pybind11

### PR DESCRIPTION
**Pull request by kfindeisen**
In addition to wrapping C++ functions defined in `coadd_utils`, this PR replaces some swig-specific code with API-based equivalents and fixes a python 2/3 compatibility issue.

### CODE REVIEW

**Comment by kfindeisen** [comment]
Hey, GitHub lets you change the PR base now! :smile:

File: .gitignore [hunk 1]
```diff
@@ -10,7 +10,6 @@ config.log
 *.pyc
 *.fits
 *_wrap.cc
```
Comment (pschella) [inline .gitignore]: This can also go.

File: python/lsst/coadd/utils/addToCoadd.cc [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+#include <pybind11/pybind11.h>
+//#include <pybind11/stl.h>
+
+#include "lsst/coadd/utils/addToCoadd.h"
+
+namespace py = pybind11;
+using namespace pybind11::literals;
+
+using namespace lsst::coadd::utils;
```
Comment (pschella) [inline python/lsst/coadd/utils/addToCoadd.cc]: Open up the namespaces instead (new convention).

File: python/lsst/coadd/utils/addToCoadd.cc [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+namespace {
+    template <typename CoaddPixelT, typename WeightPixelT>
+    void declareCoadds(py::module & mod) {
```
Comment (pschella) [inline python/lsst/coadd/utils/addToCoadd.cc]: Should really be `declareAddToCoadd`.

File: python/lsst/coadd/utils/copyGoodPixels.cc [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+#include "lsst/coadd/utils/copyGoodPixels.h"
+
+namespace py = pybind11;
+using namespace pybind11::literals;
+
+using namespace lsst::coadd::utils;
+
+namespace {
+    template <typename ImagePixelT>
+    void declareCoadds(py::module & mod) {
```
Comment (pschella) [inline python/lsst/coadd/utils/copyGoodPixels.cc]: Should be `declareCopyGoodPixels`.

File: python/lsst/coadd/utils/copyGoodPixels.cc [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+#include <pybind11/pybind11.h>
+//#include <pybind11/stl.h>
+
+#include "lsst/coadd/utils/copyGoodPixels.h"
+
+namespace py = pybind11;
+using namespace pybind11::literals;
+
+using namespace lsst::coadd::utils;
```
Comment (pschella) [inline python/lsst/coadd/utils/copyGoodPixels.cc]: open up

File: python/lsst/coadd/utils/setCoaddEdgeBits.cc [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+#include <pybind11/pybind11.h>
+//#include <pybind11/stl.h>
+
+#include "lsst/coadd/utils/setCoaddEdgeBits.h"
+
+namespace py = pybind11;
+using namespace pybind11::literals;
+
+using namespace lsst::coadd::utils;
```
Comment (pschella) [inline python/lsst/coadd/utils/setCoaddEdgeBits.cc]: open up

File: python/lsst/coadd/utils/setCoaddEdgeBits.cc [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+namespace {
+    template <typename WeightPixelT>
+    void declareCoadds(py::module & mod) {
```
Comment (pschella) [inline python/lsst/coadd/utils/setCoaddEdgeBits.cc]: should really be `declareSetCoaddEdgeBits`.

**Pull request merged**